### ISSUE TITLE
Enable DNS resolver when using EmailStr (pydantic[email] extension)

### ISSUE THREAD

**Issue by mecampbellsoup** [pydantic/pydantic#4583]
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://pydantic-docs.helpmanual.io) and still think this feature is missing

### Description

When using Pydantic w/ `email-validator` (https://github.com/JoshData/python-email-validator) I would like a way to enable DNS resolution when validating emails via `pydantic.EmailStr`.

However it is [hard-coded](https://github.com/pydantic/pydantic/blob/main/pydantic/networks.py#L726) to _not_ check deliverability of the email address. Can this be configurable instead?

For now I can just write a custom validator and not use `EmailStr` but seems like it would be useful to add this.

### Affected Components

- [ ] [Compatibility between releases](https://pydantic-docs.helpmanual.io/changelog/)
- [X] [Data validation/parsing](https://pydantic-docs.helpmanual.io/usage/models/#basic-model-usage)
- [ ] [Data serialization](https://pydantic-docs.helpmanual.io/usage/exporting_models/) - `.dict()` and `.json()`
- [ ] [JSON Schema](https://pydantic-docs.helpmanual.io/usage/schema/)
- [ ] [Dataclasses](https://pydantic-docs.helpmanual.io/usage/dataclasses/)
- [ ] [Model Config](https://pydantic-docs.helpmanual.io/usage/model_config/)
- [X] [Field Types](https://pydantic-docs.helpmanual.io/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://pydantic-docs.helpmanual.io/usage/validation_decorator/)
- [ ] [Generic Models](https://pydantic-docs.helpmanual.io/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://pydantic-docs.helpmanual.io/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://pydantic-docs.helpmanual.io/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by samuelcolvin** [comment]
Humm, I would never why these checks since they do synchronous Io, but I guess provided we had a suitable warning, there's no real harm in allowing this, opt in.

PR welcome once we're further along with v2.

**Comment by mecampbellsoup** [comment]
> Humm, I would never why these checks since they do synchronous Io, but I guess provided we had a suitable warning, there's no real harm in allowing this, opt in.

Yes, we'd like to use an async DNS adapter (to your point about this introducing a subtle yet blocking I/O call, a disaster for async apps like ours!) but just prototyping some stuff for the time being.

**Comment by samuelcolvin** [comment]
For prototyping, you can use a validator.

But note that validation and therefore validators will always be synchronous, so you'll need to to the validation elsewhere eventually.
